extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -842,22 +826,23 @@ private static void throwIfQueryIsNotSuccessful(String queryId, TaskStatusPlus s
 
   private void contextChecks(QueryContext queryContext)
   {
-    ExecutionMode executionMode = queryContext.getEnum(
-        QueryContexts.CTX_EXECUTION_MODE,
-        ExecutionMode.class,
-        null
-    );
+    ExecutionMode executionMode = queryContext.getEnum(QueryContexts.CTX_EXECUTION_MODE, ExecutionMode.class, null);
+
+    if (executionMode == null) {
+      throw InvalidInput.exception(
+          "Execution mode is not provided to the SQL statement API. Please set \"%s\" in the query context",
```

Comment:
We should also tell the user to set 
`"Please set \"%s\" to [%s] in the query context",` in the query context so he is clear of the next step to become successful.